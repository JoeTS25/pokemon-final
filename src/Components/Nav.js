import React, { useContext } from "react";
import { Link, NavLink } from "react-router-dom";
import UserContext from "../auth/UserContext";
import "./Nav.css";

function Nav({ logout }) {
    const { currentUser } = useContext(UserContext);
    console.debug("Navigation", "currentUser=", currentUser);

    function loggedinNav() {
        return (
            <ul className="navbar-nav ml-auto">
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/">Pokedex</NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/profile">Profile</NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/profile-form">Edit Profile</NavLink>
                </li>
                <li className="nav-item mr-4">
                    <Link className="nav-link" to="/" onClick={logout}>
                        Logout 
                    </Link>
                </li>
            </ul>
        );
    }

    function loggedOutNav() {
        
        return (
            <ul className="navbar-nav ml-auto">
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/">Pokedex</NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/login">Login</NavLink>
                </li>
                <li className="nav-item mr-4">
                    <NavLink className="nav-link" to="/signup">Signup</NavLink>
                </li>
                
            </ul>
        )
    }

    return (
        <nav className="Navigation navbar navbar-expand-md">
            {currentUser ? loggedinNav() : loggedOutNav()}
        </nav>
    );
}

export default Nav;
