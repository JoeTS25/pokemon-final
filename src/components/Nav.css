@font-face {
    font-family: pokemon-solid;
    src: url('../Pokemon\ Solid.ttf');
}

.Navigation{
    background-color: rgba(210, 26, 69, 0.777);
    
}

.nav-link{
    font-family: pokemon-solid;
    color: rgb(241, 219, 76);
    padding: 8px;
}

li {
    display: inline;
}




