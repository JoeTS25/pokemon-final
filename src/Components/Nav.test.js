import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Nav from "./Nav";
import { UserProvider } from "../testUtils";

it("renders without crashing", function () {
  render(
      <MemoryRouter>
        <UserProvider>
          <Nav />
        </UserProvider>
      </MemoryRouter>,
  );
});

it("matches snapshot", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider>
          <Nav />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});

it("matches snapshot when logged out", function () {
  const { asFragment } = render(
      <MemoryRouter>
        <UserProvider currentUser={null}>
          <Nav />
        </UserProvider>
      </MemoryRouter>,
  );
  expect(asFragment()).toMatchSnapshot();
});
