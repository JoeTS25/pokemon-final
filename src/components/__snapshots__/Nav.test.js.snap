// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Pokedex
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/profile"
        >
          Profile
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/profile-form"
        >
          Edit Profile
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/"
        >
          Logout
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`matches snapshot when logged out 1`] = `
<DocumentFragment>
  <nav
    class="Navigation navbar navbar-expand-md"
  >
    <ul
      class="navbar-nav ml-auto"
    >
      <li
        class="nav-item mr-4"
      >
        <a
          aria-current="page"
          class="nav-link active"
          href="/"
        >
          Pokedex
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/login"
        >
          Login
        </a>
      </li>
      <li
        class="nav-item mr-4"
      >
        <a
          class="nav-link"
          href="/signup"
        >
          Signup
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
